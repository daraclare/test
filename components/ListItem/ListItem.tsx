import React, { FC } from "react";
import styled from "styled-components";

type ListItemProps = {
  text: string;
  name: string;
  handleChange: (event: React.MouseEvent<HTMLButtonElement>) => void;
  isChecked: boolean;
  spanText?: string;
};

const StyledLi = styled.li<{ isChecked }>`
  ${({ isChecked }) => isChecked && `text-decoration: line-through;`}
`;

const StyledButton = styled.button<{ isChecked }>`
  ${({ isChecked }) => isChecked && `opacity: .5;`}
`;

const StyledSpan = styled.span`
  color: grey;
  font-size: 12px;
`;

export const ListItem: FC<ListItemProps> = ({
  text,
  handleChange,
  name,
  spanText,
  isChecked
}) => {
  return (
    <StyledLi data-testid="listItem" isChecked={isChecked}>
      {text}
      <StyledSpan data-testid="listItem-spanText"> {spanText} </StyledSpan>
      <StyledButton
        data-testid="listItem-button"
        id={name}
        onClick={handleChange}
        isChecked={isChecked}
      >
        {isChecked ? "DONE" : "IDLE"}
      </StyledButton>
    </StyledLi>
  );
};
