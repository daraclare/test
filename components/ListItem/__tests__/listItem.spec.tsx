import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { ListItem } from "../ListItem";
import { LIST_ITEM_MOCK } from "../__mocks__/listItem.mocks";

describe("Component: ListItem", () => {
  it("should render without throwing an error", () => {
    const { container } = render(<ListItem {...LIST_ITEM_MOCK} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render with correct span text", () => {
    const { getByTestId } = render(<ListItem {...LIST_ITEM_MOCK} />);
    const beerName = getByTestId("listItem-spanText");
    expect(beerName.innerHTML).toBe(` ${LIST_ITEM_MOCK.spanText} `);
  });
  it("should render with button text of IDLE", () => {
    const { getByTestId } = render(<ListItem {...LIST_ITEM_MOCK} />);
    const buttonText = getByTestId("listItem-button");
    expect(buttonText.innerHTML).toBe(`IDLE`);
  });
  it("should render with button text of DONE", () => {
    const { getByTestId } = render(
      <ListItem {...LIST_ITEM_MOCK} isChecked={true} />
    );
    const buttonText = getByTestId("listItem-button");
    expect(buttonText.innerHTML).toBe(`DONE`);
  });
});
