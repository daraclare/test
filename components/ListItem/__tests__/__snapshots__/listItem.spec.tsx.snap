// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ListItem should render without throwing an error 1`] = `
<li
  class="sc-bdVaJa ibTDPh"
  data-testid="listItem"
>
  Maris Otter Extra Pale
  <span
    class="sc-htpNat kfjJtI"
    data-testid="listItem-spanText"
  >
     
    3.3 kilograms
     
  </span>
  <button
    class="sc-bwzfXH bGgvvP"
    data-testid="listItem-button"
    id="Buzz"
  >
    IDLE
  </button>
</li>
`;
